File: DataTierGeneratorPlusTester/Vb_BuiltIn_WcfServiceApp/Test_detailContract.vb
Imports System
Imports System.Data
Imports System.Runtime.Serialization
Imports System.ServiceModel
'Imports System.ComponentModel 

'Namespace Vb_BuiltIn_WcfServiceApp 
	''' <summary>
	''' Class that defines the table operations.
	''' </summary>
	<ServiceContract()> _
	Public Interface ITest_detailService 

		''' <summary>
		''' Inserts a record into the [dbo].[test_detail] table.
		''' </summary>
		<OperationContract()> _
		Sub Insert(master_id As Int32, id As String, description As String, qty As Int32, amt As Double)

		''' <summary>
		''' Inserts a record into the [dbo].[test_detail] table.
		''' </summary>
		<OperationContract()> _
		Sub InsertInfo(ByVal info As Test_detailContract)

		''' <summary>
		''' Updates a record in the [dbo].[test_detail] table.
		''' </summary>
		<OperationContract()> _
		Sub Update(master_id As Int32, id As String, description As String, qty As Int32, amt As Double)

		''' <summary>
		''' Updates a record in the [dbo].[test_detail] table.
		''' </summary>
		<OperationContract()> _
		Sub UpdateInfo(ByVal info As Test_detailContract)

		''' <summary>
		''' Deletes a record from the [dbo].[test_detail] table by a composite primary key.
		''' </summary>
		<OperationContract()> _
		Sub Delete(master_id As Int32, id As String)

		''' <summary>
		''' Deletes a record from the [dbo].[test_detail] table by a composite primary key.
		''' </summary>
		<OperationContract()> _
		Sub DeleteInfo(ByVal info As Test_detailContract)

		''' <summary>
		''' Deletes a record from the [dbo].[test_detail] table by a foreign key.
		''' </summary>
		<OperationContract()> _
		Sub DeleteByMaster_id(master_id As Int32) 

		''' <summary>
		''' Deletes a record from the [dbo].[test_detail] table by a foreign key.
		''' </summary>
		<OperationContract()> _
		Sub DeleteInfoByMaster_id(ByVal info As Test_detailContract)

		''' <summary>
		''' Selects a single record from the [dbo].[test_detail] table.
		''' </summary>
		<OperationContract()> _
		Function SelectDS(master_id As Int32, id As String) As DataSet 

		''' <summary>
		''' Selects a single record from the [dbo].[test_detail] table.
		''' </summary>
		<OperationContract()> _
		Function SelectInfo(ByVal info As Test_detailContract) As BindingListView(Of Test_detailContract) 

		''' <summary>
		''' Selects all records from the [dbo].[test_detail] table.
		''' </summary>
		<OperationContract()> _
		Function SelectDSAll() As DataSet  

		''' <summary>
		''' Selects all records from the [dbo].[test_detail] table.
		''' </summary>
		<OperationContract()> _
		Function SelectInfoAll() As BindingListView(Of Test_detailContract)  

		''' <summary>
		''' Selects all records from the [dbo].[test_detail] table by a foreign key.
		''' </summary>
		<OperationContract()> _
		Function SelectDSByMaster_id(master_id As Int32) As DataSet 

		''' <summary>
		''' Selects all records from the [dbo].[test_detail] table by a foreign key.
		''' </summary>
		<OperationContract()> _
		Function SelectInfoByMaster_id(ByVal info As Test_detailContract) As BindingListView(Of Test_detailContract) 
	End Interface

	''' <summary>
	''' Class that stores table fields.
	''' </summary>
	<DataContract()> _
	Public Class Test_detailContract 
		''' <summary>
		''' Default constructor.  
		''' </summary>
		Public Sub New() 
		End Sub 

		''' <summary>
		''' Constructor with values.  
		''' </summary>
		Public Sub New( _
			master_id As Int32, _
			id As String, _
			description As String, _
			qty As Int32, _
			amt As Double _
		)
			_master_id = master_id
			_id = id
			_description = description
			_qty = qty
			_amt = amt
		End Sub

		' Private variables for columns in the test_detail table.
		Private  _master_id As Int32
		Private  _id As String
		Private  _description As String
		Private  _qty As Int32
		Private  _amt As Double

		' Public properties for columns in the test_detail table.

		'<DataObjectFieldAttribute(True, False, False)> _
		''' <summary>
		''' master_id  
		''' </summary>
		<DataMember()> _
		Public Property master_id As Int32
			Get 
				Return _master_id 
			End Get
			Set(ByVal Value As Int32) 
				_master_id = Value
			End Set
		End Property

		'<DataObjectFieldAttribute(True, False, False)> _
		''' <summary>
		''' id  
		''' </summary>
		<DataMember()> _
		Public Property id As String
			Get 
				Return _id 
			End Get
			Set(ByVal Value As String) 
				_id = Value
			End Set
		End Property

		'<DataObjectFieldAttribute(False, False, False)> _
		''' <summary>
		''' description  
		''' </summary>
		<DataMember()> _
		Public Property description As String
			Get 
				Return _description 
			End Get
			Set(ByVal Value As String) 
				_description = Value
			End Set
		End Property

		'<DataObjectFieldAttribute(False, False, False)> _
		''' <summary>
		''' qty  
		''' </summary>
		<DataMember()> _
		Public Property qty As Int32
			Get 
				Return _qty 
			End Get
			Set(ByVal Value As Int32) 
				_qty = Value
			End Set
		End Property

		'<DataObjectFieldAttribute(False, False, False)> _
		''' <summary>
		''' amt  
		''' </summary>
		<DataMember()> _
		Public Property amt As Double
			Get 
				Return _amt 
			End Get
			Set(ByVal Value As Double) 
				_amt = Value
			End Set
		End Property
	End Class
'End Namespace

File: DataTierGeneratorPlusTester/Cs_BuiltIn_WcfServiceApp/Test_detailContract.cs
using System;
using System.Data;
using System.Runtime.Serialization;
using System.ServiceModel;
//using System.ComponentModel;

namespace Cs_BuiltIn_WcfServiceApp 
{
	/// <summary>
	/// Class that defines the table operations.
	/// </summary>
	[ServiceContract]
	public interface ITest_detailService 
	{

		/// <summary>
		/// Inserts a record into the [dbo].[test_detail] table.
		/// </summary>
		[OperationContract]
		void Insert(Int32 master_id, String id, String description, Int32 qty, Double amt);

		/// <summary>
		/// Inserts a record into the [dbo].[test_detail] table.
		/// </summary>
		[OperationContract]
		void InsertInfo(Test_detailContract info);

		/// <summary>
		/// Updates a record in the [dbo].[test_detail] table.
		/// </summary>
		[OperationContract]
		void Update(Int32 master_id, String id, String description, Int32 qty, Double amt);

		/// <summary>
		/// Updates a record in the [dbo].[test_detail] table.
		/// </summary>
		[OperationContract]
		void UpdateInfo(Test_detailContract info);

		/// <summary>
		/// Deletes a record from the [dbo].[test_detail] table by a composite primary key.
		/// </summary>
		[OperationContract]
		void Delete(Int32 master_id, String id);

		/// <summary>
		/// Deletes a record from the [dbo].[test_detail] table by a composite primary key.
		/// </summary>
		[OperationContract]
		void DeleteInfo(Test_detailContract info);

		/// <summary>
		/// Deletes a record from the [dbo].[test_detail] table by a foreign key.
		/// </summary>
		[OperationContract]
		void DeleteByMaster_id(Int32 master_id);

		/// <summary>
		/// Deletes a record from the [dbo].[test_detail] table by a foreign key.
		/// </summary>
		[OperationContract]
		void DeleteInfoByMaster_id(Test_detailContract info);

		/// <summary>
		/// Selects a single record from the [dbo].[test_detail] table.
		/// </summary>
		[OperationContract]
		DataSet SelectDS(Int32 master_id, String id);

		/// <summary>
		/// Selects a single record from the [dbo].[test_detail] table.
		/// </summary>
		[OperationContract]
		BindingListView<Test_detailContract> SelectInfo(Test_detailContract info);

		/// <summary>
		/// Selects all records from the [dbo].[test_detail] table.
		/// </summary>
		[OperationContract]
		DataSet SelectDSAll();

		/// <summary>
		/// Selects all records from the [dbo].[test_detail] table.
		/// </summary>
		[OperationContract]
		BindingListView<Test_detailContract> SelectInfoAll();

		/// <summary>
		/// Selects all records from the [dbo].[test_detail] table by a foreign key.
		/// </summary>
		[OperationContract]
		DataSet SelectDSByMaster_id(Int32 master_id);

		/// <summary>
		/// Selects all records from the [dbo].[test_detail] table by a foreign key.
		/// </summary>
		[OperationContract]
		BindingListView<Test_detailContract> SelectInfoByMaster_id(Test_detailContract info);
	}

	/// <summary>
	/// Class that stores table fields.
	/// </summary>
	[DataContract]
	public class Test_detailContract 
	{
		/// <summary>
		/// Default constructor.  
		/// </summary>
		public Test_detailContract() {}

		/// <summary>
		/// Constructor with values.  
		/// </summary>
		public Test_detailContract(
			Int32 master_id,
			String id,
			String description,
			Int32 qty,
			Double amt
		)
		{
			_master_id = master_id;
			_id = id;
			_description = description;
			_qty = qty;
			_amt = amt;
		}

		/// Private variables for columns in the test_detail table.
		private Int32 _master_id;
		private String _id;
		private String _description;
		private Int32 _qty;
		private Double _amt;

		/// Public properties for columns in the test_detail table.

		/// <summary>
		/// master_id  
		/// </summary>
		[DataMember]
		//[DataObjectFieldAttribute(true, false, false)]
		public Int32 master_id
		{
			get { return _master_id; }
			set { _master_id = value; }
		}

		/// <summary>
		/// id  
		/// </summary>
		[DataMember]
		//[DataObjectFieldAttribute(true, false, false)]
		public String id
		{
			get { return _id; }
			set { _id = value; }
		}

		/// <summary>
		/// description  
		/// </summary>
		[DataMember]
		//[DataObjectFieldAttribute(false, false, false)]
		public String description
		{
			get { return _description; }
			set { _description = value; }
		}

		/// <summary>
		/// qty  
		/// </summary>
		[DataMember]
		//[DataObjectFieldAttribute(false, false, false)]
		public Int32 qty
		{
			get { return _qty; }
			set { _qty = value; }
		}

		/// <summary>
		/// amt  
		/// </summary>
		[DataMember]
		//[DataObjectFieldAttribute(false, false, false)]
		public Double amt
		{
			get { return _amt; }
			set { _amt = value; }
		}
	}
}
